temporal-sdk/src/test/java/io/temporal/internal/worker/WorkflowSlotsSmallSizeTests.java

```diff
@@ -116,24 +106,9 @@ public void tearDown() {
         localActivitySlotSupplier.releasedCount.get());
   }
 
-  private void assertWorkerSlotCount(int worker, int activity, int localActivity) {
-    try {
-      // There can be a delay in metrics emission, another option if this
-      // is too flaky is to poll the metrics.
-      Thread.sleep(100);
-    } catch (InterruptedException e) {
-      throw new RuntimeException(e);
-    }
-    reporter.assertGauge(
-        MetricsType.WORKER_TASK_SLOTS_AVAILABLE, getWorkerTags("WorkflowWorker"), worker);
-    // All slots should be available
-    reporter.assertGauge(
-        MetricsType.WORKER_TASK_SLOTS_AVAILABLE, getWorkerTags("ActivityWorker"), activity);
-    // All slots should be available
-    reporter.assertGauge(
-        MetricsType.WORKER_TASK_SLOTS_AVAILABLE,
-        getWorkerTags("LocalActivityWorker"),
-        localActivity);
+  private void assertCurrentUsedCount(int activity, int localActivity) {
+    assertEquals(activity, activityTaskSlotSupplier.currentUsedSet.size());
```

Comment:
I removed the workflow slot check because when it was released could be pretty timing dependent, and that's better covered by other tests anyway.